static/app/components/performance/searchBar.tsx

```diff
@@ -227,14 +229,39 @@ function SearchBar(props: SearchBarProps) {
           searchSubstring={searchString}
           loading={loading}
           items={searchResults}
-          onClick={handleSearch}
-          onIconClick={navigateToTransactionSummary}
+          onClick={handleChooseItem}
+          onIconClick={handleClickItemIcon}
         />
       )}
     </Container>
   );
 }
 
+const encodeItemToValue = (item: DataItem) => {
+  return `${item.transaction}:${item.project_id}`;
+};
+
+const decodeValueToItem = (value: string): DataItem => {
+  const lastIndex = value.lastIndexOf(':');
+
+  return {
+    project_id: parseInt(value.slice(lastIndex + 1), 10),
+    transaction: value.slice(0, lastIndex),
+  };
+};
+
+interface DataItem {
+  project_id: number;
+  transaction: string;
+  'count()'?: number;
+}
+
+const wrapQueryInWildcards = (query: string) => {
+  const prependedChar = query[0] === '*' ? '' : '*';
+  const appendedChar = query[query.length - 1] === '*' ? '' : '*';
```

Comment:
😅 this is the code that was there, I only moved it. I didn't want to change every single thing! In any cause. In cany case, since you called it out, I changed the function a bit to make the intent crystal clear

Reply:
Sorry, that is definitely fair! No need to touch it if its not part of the PR, but I suppose why not!

Reply:
I feel pretty good about it since the unit tests are there 👍🏻